sdk/cosmos/azure_data_cosmos/src/clients/cosmos_client.rs

```diff
@@ -96,6 +96,37 @@ impl CosmosClient {
         })
     }
 
+    /// Creates a new CosmosClient, using a connection string.
+    ///
+    /// # Arguments
+    ///
+    /// * `connection_string` - the connection string to use for the client, e.g. AccountEndpoint=https://accountname.documents.azure.com:443/‌​;AccountKey=accountk‌​ey
+    /// * `options` - Optional configuration for the client.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use azure_data_cosmos::CosmosClient;
+    /// use azure_core::credentials::Secret;
+    ///
+    /// let client = CosmosClient::with_connection_string(
+    ///     "AccountEndpoint=https://accountname.documents.azure.com:443/‌​;AccountKey=accountk‌​ey",
+    ///     None)
+    ///     .unwrap();
+    /// ```
+    #[cfg(feature = "key_auth")]
+    #[allow(rustdoc::bare_urls)]
```

Comment:
I _think_ if the connection string is in backticks, this lint won't trigger anymore. The reason it's triggering is that it's seeing a plain URL, which rustdoc will NOT turn in to a link by default (so the lint is warning you that you really should make it a link if it's a URL someone might navigate to). However, in our case, it's part of an example Connection String. Wrapping it in backticks (`like this`) will tell rustdoc that it's a verbatim string that shouldn't be parsed as markdown and avoid the lint entirely.